gradle/dependencies.gradle

```diff
@@ -15,367 +15,386 @@
+                                it.name != 'maven-model' &&
+                                it.name != 'maven-model-builder' &&
+                                it.name != 'maven-artifact' &&
+                                it.name != 'maven-aether-provider' &&
+                                it.group != 'org.sonatype.aether'
+                        }
                     }
                 }
-                withVariant('runtime') {
-                    withDependencies { it.removeAll { it.group == 'org.sonatype.sisu' }}
-                }
             }
-            withModule('org.apache.maven.wagon:wagon-provider-api') {
-                withVariant('default') {
-                    withDependencies { it.findAll { it.name == 'plexus-utils' }.each { changeVersionInMetadata it, '2.1' }}
-                }
-                withVariant('runtime') {
-                    withDependencies { it.findAll { it.name == 'plexus-utils' }.each { changeVersionInMetadata it, '2.1' }}
+            withModule('com.amazonaws:aws-java-sdk-core') {
+                allVariants {
+                    withDependencies { it.removeAll { it.name == 'jackson-dataformat-cbor' } }
                 }
             }
-            withModule('org.apache.xbean:xbean-reflect') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group == 'commons-logging' || it.group == 'log4j' }}
-                }
-                withVariant('runtime') {
-                    withDependencies { it.removeAll { it.group == 'commons-logging' || it.group == 'log4j' }}
+            withModule('org.eclipse.jgit:org.eclipse.jgit') {
+                allVariants {
+                    withDependencies { it.removeAll { it.group == 'com.googlecode.javaewah' }}
                 }
             }
-            withModule('org.codehaus.plexus:plexus-container-default') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group == 'com.google.collections' || it.group == 'junit' || it.group == 'org.apache.xbean' }}
-                }
-                withVariant('runtime') {
-                    withDependencies { it.removeAll { it.group == 'com.google.collections' || it.group == 'junit' || it.group == 'org.apache.xbean' }}
+            withModule('org.apache.maven.wagon:wagon-http-shared4') {
+                allVariants {
+                    withDependencies { it.removeAll { it.group == 'org.jsoup' }}
                 }
             }
             withModule('org.sonatype.aether:aether-connector-wagon') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group == 'org.sonatype.sisu' }}
-                }
-                withVariant('runtime') {
-                    withDependencies { it.removeAll { it.group == 'org.sonatype.sisu' }}
+                allVariants {
+                    withDependencies { it.removeAll { it.group == 'org.sonatype.sisu' } }
```

Comment:
We have a `rejectAll` constraint for `org.sonatype.sisu`. That's probably a better place for it.

Reply:
Follow up: https://github.com/gradle/gradle/issues/4230